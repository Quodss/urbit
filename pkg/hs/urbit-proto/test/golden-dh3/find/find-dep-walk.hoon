=/  f
  ^-  $-(@ $)
  |=  a
  {@ @}
=/  c
  ^-  {a|@ b|(f a)}
  [1 2 3]
b.c
