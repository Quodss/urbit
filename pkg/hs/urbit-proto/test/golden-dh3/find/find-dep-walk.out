PARSED

=/  f
  ^-  $-(@ $)
  |=(a {@ @})
=/  c
  ^-  {a|@ b|(f a)}
  [1 2 3]
b.c

CODE

=+  ^=  f
    ^-  $-(@ $)
    |=(a {@ @})
=+  ^=  c
    ^-  {a=@ b=(f a)}
    [1 2 3]
b.c

COLD

=+  f=|=(_ {@ @})
=+  c=[1 2 3]
+7

TYPE

=>  :-  :+  :-  0
            ^=  f
            =>  0
            |=(_ {@ @})
          0
        ^=  f
        =>  0
        |=(_ {@ @})
    1
{@ @}

BASE

[2 3]